CI broken linting on Go 1.5

Travis builds failing because of https://github.com/golang/lint/issues/246

```
# github.com/golang/lint
../../golang/lint/lint.go:249: undefined: types.ImportMode
make: *** [install-travis] Error 2
The command "make install-travis" failed and exited with 2 during .
Your build has been stopped.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,8 @@
 language: go
 
 go:
-  - 1.5.4
   - 1.6.2
+  - 1.7.1
 
 install:
   - make install-travis

--- a/pkg/test/git.go
+++ b/pkg/test/git.go
@@ -86,7 +86,7 @@ func (f *FakeGit) GetInfo(repo string) *api.SourceInfo {
 	}
 }
 
-// Creates a git directory with one unlikely but possible commit hash
+// CreateLocalGitDirectory creates a git directory with one unlikely but possible commit hash
 func CreateLocalGitDirectory(t *testing.T) string {
 	dir, err := ioutil.TempDir(os.TempDir(), "gitdir-s2i-test")
 	if err != nil {

--- a/pkg/test/tar.go
+++ b/pkg/test/tar.go
@@ -25,8 +25,17 @@ type FakeTar struct {
 func (f *FakeTar) Copy() *FakeTar {
 	f.lock.Lock()
 	defer f.lock.Unlock()
-	n := *f
-	return &n
+	// copy everything except .lock...
+	n := &FakeTar{
+		CreateTarBase:    f.CreateTarBase,
+		CreateTarDir:     f.CreateTarDir,
+		CreateTarResult:  f.CreateTarResult,
+		CreateTarError:   f.CreateTarError,
+		ExtractTarDir:    f.ExtractTarDir,
+		ExtractTarReader: f.ExtractTarReader,
+		ExtractTarError:  f.ExtractTarError,
+	}
+	return n
 }
 
 // CreateTarFile creates a new fake UNIX tar file
@@ -38,7 +47,7 @@ func (f *FakeTar) CreateTarFile(base, dir string) (string, error) {
 	return f.CreateTarResult, f.CreateTarError
 }
 
-// ExtractTarStream streams a content of fake tar
+// ExtractTarStreamWithLogging streams a content of fake tar
 func (f *FakeTar) ExtractTarStreamWithLogging(dir string, reader io.Reader, logger io.Writer) error {
 	f.lock.Lock()
 	defer f.lock.Unlock()

